c10/util/flat_hash_map.h

```diff
@@ -233,12 +233,14 @@ template <
     typename T,
     typename FindKey,
     typename ArgumentHash,
-    typename Hasher,
+    typename DetailHasher,
```

Comment:
I'm not sure I understand how this change is related to enabling Vulka on Windows

Reply:
This is related to the a VS bug(feature?). Link [here](https://developercommunity.visualstudio.com/t/nested-templates-causing-compilation-error-when-th/1397858?from=email#T-ND1399286)

Reply:
I couldn't find a way to control `/permissive` per file with a #pragma either. Does another trick come to anyone's mind here?